app/components/pdf-document.js

```diff
@@ -0,0 +1,264 @@
+/* globals PDFJS Promise */
+import Ember from 'ember';
+const get = Ember.get;
+const set = Ember.set;
+const bind = Ember.run.bind;
+const $window = Ember.$(window);
+
+export default Ember.Component.extend({
+
+  /**
+  * The `classNames` of the document container.
+  *
+  * @property
+  * @default
+  */
+  classNames: ['pdf-document-container'],
+
+  /**
+  * The `src` of the document being requested.
+  *
+  * @property
+  * @default Empty string
+  */
+  src: '',
+
+  /**
+  * The property storing the document.
+  *
+  * @property
+  * @default null
+  */
+  docObject: null,
+
+  /**
+  * Hook that tears down scroll binding
+  *
+  * @method  willDestroyElement
+  * @return void
+  */
+  willDestroyElement() {
+    $window.off('scroll.pdf');
```

Comment:
if this can be an arbitrary string, please make it particular to the component instance, even if it needs to be machine computed.

imagine showing multiple docs side-by-side... since you're proxying event handling on the `window` object -- any subscriber tearing down this named event would blow it away for any other subscriber, e.g. a second instance of this component.